build: buildx 0.24.0 release notes (#22689)
docker/docs: content/manuals/build/release-notes.md

This page contains information about the new features, improvements, and bug
fixes in [Docker Buildx](https://github.com/docker/buildx).

## 0.24.0

{{< release-date date="2025-05-21" >}}

> The full release note for this release is available

Reviewer: ```suggestion
The full release notes for this release are available
```

[on GitHub](https://github.com/docker/buildx/releases/tag/v0.24.0).

### Enhancements

> - New `type` attribute added to `variable` block in Bake to allow explicit typing of variables. [docker/buildx#3167](https://github.com/docker/buildx/pull/3167), [docker/buildx#3189](https://github.com/docker/buildx/pull/3189), [docker/buildx#3198](https://github.com/docker/buildx/pull/3198)

Reviewer: ```suggestion
- New `type` attribute added to the `variable` block in Bake to allow explicit typing of variables. [docker/buildx#3167](https://github.com/docker/buildx/pull/3167), [docker/buildx#3189](https://github.com/docker/buildx/pull/3189), [docker/buildx#3198](https://github.com/docker/buildx/pull/3198)
```

> - New `--finalize` flag added to `history export` command to finalize build traces before exporting. [docker/buildx#3152](https://github.com/docker/buildx/pull/3152)

Reviewer: ```suggestion
- New `--finalize` flag added to the `history export` command to finalize build traces before exporting. [docker/buildx#3152](https://github.com/docker/buildx/pull/3152)
```

- Compose compatibility has been updated to v2.6.3. [docker/buildx#3191](https://github.com/docker/buildx/pull/3191), [docker/buildx#3171](https://github.com/docker/buildx/pull/3171)

### Bug fixes

- Fix issue where some builds may leave behind temporary files after completion. [docker/buildx#3133](https://github.com/docker/buildx/pull/3133)

> - Fix wrong image ID returned when building with docker and containerd-snapshotter enabled. [docker/buildx#3136](https://github.com/docker/buildx/pull/3136)

Reviewer: ```suggestion
- Fix wrong image ID returned when building with Docker when containerd-snapshotter is enabled. [docker/buildx#3136](https://github.com/docker/buildx/pull/3136)
```
